src/App.tsx

```diff
@@ -35,17 +43,56 @@ export const theme = createMuiTheme({
 });
 
 function App() {
+  const [initializing, setInitializing] = useState(true);
+  const [namespaces, setNamespaces] = useState<INamespace[]>([]);
+  const [selectedNamespace, setSelectedNamespace] = useState<string>('');
+
+  useEffect(() => {
+    fetch('/api/v1/namespaces')
+      .then(async (response) => {
+        if (response.ok) {
+          const ns: INamespace[] = await response.json();
+          setNamespaces(ns);
+          const storageItem = window.localStorage.getItem(
+            NAMESPACE_LOCALSTORAGE_KEY
+          );
+          if (storageItem) {
+            setSelectedNamespace(storageItem);
+          } else if (ns.length !== 0) {
+            setSelectedNamespace(ns[0].name);
```

Comment:
Do we know for sure that the first one will always be the `default` namespace, as returned by the FireFly API? If the order of this array is unpredictable, will that cause unintended behavior?